src/app/lightning/update-ln-payments.ts

```diff
@@ -0,0 +1,93 @@
+import { LndService } from "@services/lnd"
+import { baseLogger } from "@services/logger"
+import { LnPaymentsRepository } from "@services/mongoose/ln-payments"
+
+export const updateLnPayments = async (): Promise<true | ApplicationError> => {
+  let incompleteLnPayments = await LnPaymentsRepository().listIncomplete()
+  if (incompleteLnPayments instanceof Error) return incompleteLnPayments
+
+  const lndService = LndService()
+  if (lndService instanceof Error) return lndService
+
+  const lndListMethods = [lndService.listSettledPayments, lndService.listFailedPayments]
+  for (const listPaymentsFn of lndListMethods) {
+    incompleteLnPayments = await fetchAndUpdatePayments({
+      incompleteLnPayments,
+      listPaymentsFn,
+    })
+    if (incompleteLnPayments instanceof Error) return incompleteLnPayments
+    if (incompleteLnPayments.length == 0) return true
```

Comment:
should we return here? this is within a loop and we're going there twice, and shouldn't return if the other loop has not completed? because we may have to return an error but the first iteration of the loop may already have returned

Reply:
The `fetchAndUpdatePayments` takes an initial `incompleteLnPayments` array and returns a pared down version of `incompleteLnPayments` so this return is for if there are no more incomplete payments from our `lnPayments` collection to be updated. In that case we don't need to go to `lnd` for anything anymore regardless of the method we're currently in.